lib/src/modules/gates.dart

```diff
@@ -705,6 +707,11 @@ class IndexGate extends Module with InlineSystemVerilog {
 
   /// Executes the functional behavior of this gate.
   void _execute() {
+    if (_original.width == 1) {
+      selection.put(_original.value);
+      return;
+    }
+
     if (_index.hasValidValue()) {
       final indexVal = _index.value.toInt();
       selection.put(_original.value.getRange(indexVal, indexVal + 1));
```

Comment:
I think we need to drive `x` if `indexVal` is out of range, I guess a miss from before, but one that your new tests will cover (with above modification)